### ISSUE TITLE
unet_learner causes "Kernel died unexpectedly."

### ISSUE THREAD

**Issue by matthewjur1** [fastai/fastai#2273]
This code runs:
model = torchvision.models.segmentation.deeplabv3_resnet50(num_classes=2, aux_loss=True)
learn = Learner(data, model, loss_func=loss_func, opt_func=optim.Adamax)

This code causes error /w everything else the exact same:
arch = torchvision.models.resnet50
learn = unet_learner(data, arch, loss_func=dice_loss, opt_func=optim.Adamax)

Ubuntu 18.04.1 LTS
FastAI v1.0.57

**Comment by matthewjur1** [comment]
def dice_loss(logits, labels, eps=1e-8, reduction='mean'):
    logits = F.softmax(logits, dim=1)
    labels = torch.eye(2)[labels.squeeze(1)].permute(0,3,1,2)
    labels = labels.float().to(defaults.device)

    i = (logits * labels).sum(dim=[2,3])
    u = (logits + labels).sum(dim=[2,3])
    loss = (1 - 2 * i / (u + eps)).mean(dim=1)

    return loss.mean() if reduction != 'none' else loss

def loss_func(logits, labels, reduction='mean'):
    return (dice_loss(logits['out'], labels, reduction=reduction) + 
        .3 * dice_loss(logits['aux'], labels, reduction=reduction))
